### PR TITLE
debian package building added

### PR DESCRIPTION
**Pull request by navjotcis**
I have added debian package building with the reference of https://git.bitmessage.org/citizenaspirant/PyBitmessage/commit/5f955ba52f10c6167e82ec870cbb327a6ca1de2d

### CODE REVIEW

**Comment by PeterSurda** [comment]
I tested, it builds, but haven't tried installing.

**Comment by PeterSurda** [comment]
I would say I merge this and if it doesn't run, fix it separately, because having this in will allow me to add dpkg to buildbot.

**Review by PeterSurda — APPROVED** [review]

**Comment by g1itch** [comment]
Why don't you put this into packages dir?

**Comment by PeterSurda** [comment]
I don't know how to integrated it into `dpkg-buildpackage` if I move the files.

File: debian/rules
Comment (g1itch) [inline debian/rules:10]: Did you try --sourcedirectory=.. here?

Comment (navjotcis) [reply debian/rules:10]: image
I have done some changes, please look

Comment (PeterSurda) [reply debian/rules:10]: add the argument to the existing command, don't add a new command

Comment (navjotcis) [reply debian/rules:10]: ok now please review it
image

Comment (g1itch) [reply debian/rules:10]: @navjotcis I meant after moving the stuff to the packages dir.

Comment (g1itch) [reply debian/rules:10]: Well, I checked: --sourcedirectory=.. not works.

**Comment by g1itch** [comment]
That's what I expected: https://travis-ci.org/github/g1itch/PyBitmessage/builds/735505903

File: debian/rules
Comment (navjotcis) [reply debian/rules]: yes it is failing the build

**Comment by navjotcis** [comment]
Now the build is successfully passing https://buildbot.bitmessage.org/#/builders/14/builds/161

**Comment by g1itch** [comment]
Also, please, look into informational part. Why empty fields? Why such weird version and unstable mark? Should python-msgpack really be a requirement? Who is going to be maintainer and how it will be maintained? Changelog may be probably generated by sphinx.

**Comment by navjotcis** [comment]
> Also, please, look into informational part. Why empty fields? Why such weird version and unstable mark? Should python-msgpack really be a requirement? Who is going to be maintainer and how it will be maintained? Changelog may be probably generated by sphinx.

so I think I should use stable version like you have used `bitmessage (0.6.3.2) stable; urgency=medium`

**Comment by g1itch** [comment]
> > Also, please, look into informational part. Why empty fields? Why such weird version and unstable mark? Should python-msgpack really be a requirement? Who is going to be maintainer and how it will be maintained? Changelog may be probably generated by sphinx.
> 
> so I think I should use stable version like you have used `bitmessage (0.6.3.2) stable; urgency=medium`

You can get actual version by `python setup.py -V` but it's not the point. For me the real problem with this PR is that it tends to become another unusable piece of garbage like `fabfile`, `android_instruction.rst`, `src/bitmessagekivy`, `src/kivymd`, `src/navigationdrawer` and like `docs` was before I fixed the documentation building process. You have some additional scripts in `buildbot/dpkg` task, including one, changing the version, which is needed for deb building. But your README says "Clone the source and run dpkg-buildpackage" and that's not gonna work for a general end user.

**Comment by g1itch** [comment]
> That's what I expected: https://travis-ci.org/github/g1itch/PyBitmessage/builds/735505903

Amendment: something like that, but demonstrating the build of non-empty and functional deb.

**Comment by g1itch** [comment]
> > That's what I expected: https://travis-ci.org/github/g1itch/PyBitmessage/builds/735505903
> 
> Amendment: something like that, but demonstrating the build of non-empty and functional deb.

Like this: https://travis-ci.org/github/g1itch/PyBitmessage/jobs/736037581

As I suspected all that `debian` dir indeed is garbage. stdeb.cfg is enough. If someone needs that dir for some reason, it's generated by `python setup.py --command-packages=stdeb.command debianize`.

**Comment by PeterSurda** [comment]
@g1itch  Can you make a minimal PR from your `debian` branch? Then we can close this. No need to modify `travis.yml`, I want the builds and the tests to be separate. As you can see buildbot's dpkg builder now succeeds, so it just needs fine tuning.

**Comment by PeterSurda** [comment]
Superseded by #1686 .

**Pull request closed without merging**
